cmd/list.go

```diff
@@ -82,62 +98,198 @@ func list(cmd *cobra.Command) {
+				// TODO
+			}
+
+			row = append(row, value)
+		}
+		data = append(data, row)
+	}
+
 	if jsonOutput {
-		utils.DisplayJsonOutput(listResp.Payload)
+		// TODO Fix json output
+		utils.DisplayJsonOutput(data)
 	} else {
-		displayListPayload(listResp.Payload)
+		utils.DisplayTable(header, data)
 	}
 }
 
-func displayListPayload(p *operations.GetWorkflowsOKBody) {
-	header := []interface{}{
-		"NAME",
-		"RUN_NUMBER",
-		"CREATED",
-		"STARTED",
-		"ENDED",
-		"STATUS",
+func displayWorkflowDuration(workflow *operations.GetWorkflowsOKBodyItemsItems0) string {
+	runStartedAt := workflow.Progress.RunStartedAt
+	runFinishedAt := workflow.Progress.RunFinishedAt
+	if runStartedAt == nil {
+		return "-"
+	}
+	startTime := utils.FromIsoToTimestamp(*runStartedAt)
+	var endTime time.Time
+	if runFinishedAt != nil {
+		endTime = utils.FromIsoToTimestamp(*runFinishedAt)
+	} else {
+		endTime = time.Now()
 	}
-	var rows [][]interface{}
+	duration := endTime.Sub(startTime).Round(time.Second).Seconds()
+	return fmt.Sprintf("%g", duration)
+}
 
-	for _, workflow := range p.Items {
-		var row []interface{}
-		workflowNameAndRunNumber := strings.SplitN(workflow.Name, ".", 2)
-		row = append(
-			row,
-			workflowNameAndRunNumber[0],
-			workflowNameAndRunNumber[1],
-			workflow.Created,
-			displayOptionalField(workflow.Progress.RunStartedAt),
-			displayOptionalField(workflow.Progress.RunFinishedAt),
-			workflow.Status,
-		)
-		rows = append(rows, row)
-	}
-
-	utils.DisplayTable(header, rows)
+func getWorkflowNameAndRunNumber(workflowName string) (string, string) {
+	workflowNameAndRunNumber := strings.SplitN(workflowName, ".", 2)
+	if len(workflowNameAndRunNumber) < 2 {
+		return workflowName, ""
+	}
+	return workflowNameAndRunNumber[0], workflowNameAndRunNumber[1]
 }
 
-func displayOptionalField(value *string) string {
+func buildListHeaders(
+	runType string,
+	verbose, includeWorkspaceSize, includeProgress, includeDuration bool,
+) map[string][]any {
+	verboseHeaders := []any{"id", "user"}
+	workspaceSizeHeader := []any{"size"}
+	progressHeader := []any{"progress"}
+	durationHeader := []any{"duration"}
+	headers := map[string][]any{
+		"batch": {"name", "run_number", "created", "started", "ended", "status"},
+		"interactive": {
+			"name",
+			"run_number",
+			"created",
+			"session_type",
+			"session_uri",
+			"session_status",
+		},
+	}
+
+	if verbose {
+		headers[runType] = append(headers[runType], verboseHeaders...)
+	}
+	if verbose || includeWorkspaceSize {
+		headers[runType] = append(headers[runType], workspaceSizeHeader...)
+	}
+	if verbose || includeProgress {
+		headers[runType] = append(headers[runType], progressHeader...)
+	}
+	if verbose || includeDuration {
+		headers[runType] = append(headers[runType], durationHeader...)
+	}
+
+	return headers
+}
+
+func displayOptionalStringField(value *string) string {
 	if value == nil {
 		return "-"
 	}
 	return *value
 }
+
+func displayOptionalFloatField(value float64) string {
```

Comment:
As before, this function does not display the value passed, so we should probably not call it `display...`. WDYT?